compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java

```diff
@@ -831,6 +841,20 @@ public void visit(BLangInvocation.BLangActionInvocation actionInvocationExpr) {
 
     @Override
     public void visit(BLangTypeInit typeInitExpr) {
+        BSymbol initInvocationSymbol = typeInitExpr.initInvocation.symbol;
+        if (initInvocationSymbol != null && !isIsolated(initInvocationSymbol)) {
+            inferredIsolated = false;
+
+            if (isInIsolatedFunction(env.enclInvokable)) {
+                dlog.error(typeInitExpr.pos, DiagnosticCode.INVALID_NON_ISOLATED_INIT_EXPRESSION_IN_ISOLATED_FUNCTION);
+            } else if (isRecordFieldDefaultValue(env.enclType)) {
+                // TODO: 9/16/20 make this error once stdlibs are migrated
```

Comment:
Better to create a git issue for this. Otherwise we might forget these with time.

Reply:
Created https://github.com/ballerina-platform/ballerina-lang/issues/25850. Thanks!